collated/test/A0141021H.md

```diff
@@ -0,0 +1,537 @@
+# A0141021H
+###### \java\seedu\whatnow\logic\LogicManagerTest.java
+``` java
+package seedu.whatnow.logic;
+
+import com.google.common.eventbus.Subscribe;
+
+import seedu.whatnow.commons.core.EventsCenter;
+import seedu.whatnow.commons.events.model.WhatNowChangedEvent;
+import seedu.whatnow.commons.events.ui.JumpToListRequestEvent;
+import seedu.whatnow.commons.events.ui.ShowHelpRequestEvent;
+import seedu.whatnow.logic.Logic;
+import seedu.whatnow.logic.LogicManager;
+import seedu.whatnow.logic.commands.*;
+import seedu.whatnow.model.Model;
+import seedu.whatnow.model.ModelManager;
+import seedu.whatnow.model.ReadOnlyWhatNow;
+import seedu.whatnow.model.WhatNow;
+import seedu.whatnow.model.tag.Tag;
+import seedu.whatnow.model.tag.UniqueTagList;
+import seedu.whatnow.model.task.*;
+import seedu.whatnow.storage.StorageManager;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static seedu.whatnow.commons.core.Messages.*;
+
+public class LogicManagerTest {
+
+    /**
+     * See https://github.com/junit-team/junit4/wiki/rules#temporaryfolder-rule
+     */
+    @Rule
+    public TemporaryFolder saveFolder = new TemporaryFolder();
+
+    private Model model;
+    private Logic logic;
+
+    //These are for checking the correctness of the events raised
+    private ReadOnlyWhatNow latestSavedWhatNow;
+    private boolean helpShown;
+    private int targetedJumpIndex;
+
+    @Subscribe
+    private void handleLocalModelChangedEvent(WhatNowChangedEvent abce) {
+        latestSavedWhatNow = new WhatNow(abce.data);
+    }
+
+    @Subscribe
+    private void handleShowHelpRequestEvent(ShowHelpRequestEvent she) {
+        helpShown = true;
+    }
+
+    @Subscribe
+    private void handleJumpToListRequestEvent(JumpToListRequestEvent je) {
+        targetedJumpIndex = je.targetIndex;
+    }
+
+    @Before
+    public void setup() {
+        model = new ModelManager();
+        String tempWhatNowFile = saveFolder.getRoot().getPath() + "TempWhatNow.xml";
+        String tempPreferencesFile = saveFolder.getRoot().getPath() + "TempPreferences.json";
+        logic = new LogicManager(model, new StorageManager(tempWhatNowFile, tempPreferencesFile));
+        EventsCenter.getInstance().registerHandler(this);
+
+        latestSavedWhatNow = new WhatNow(model.getWhatNow()); // last saved assumed to be up to date before.
+        helpShown = false;
+        targetedJumpIndex = -1; // non yet
+    }
+
+    @After
+    public void teardown() {
+        EventsCenter.clearSubscribers();
+    }
+
+    @Test
+    public void execute_invalid() throws Exception {
```

Comment:
test name should also include what is the expected outcome. recollect that the test name has 3 parts.. blah... blah..